uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/absence-tab/components/contract-entitlements.js

```diff
@@ -0,0 +1,95 @@
+/* eslint-env amd */
+
+define([
```

Comment:
I reduced the code between sortBy and map as you suggested. I also added the formatDate function. Deleted unnecessary vars and deps. However I am not sure I agree with those: 

1) 
```js
vm.absenceTypes = []; // tells me absence types loaded, but none found
absenceTypes = null; // tells me there are no absence types loaded yet
// this also allows to not to have those unnecessary "absenceTypeLoading" like props
```
2) `absences` and `absence types` must be synced in terms of order, otherwise there is a risk of showing wrong data.